components/org.wso2.carbon.identity.application.authenticator.fido2/src/main/java/org/wso2/carbon/identity/application/authenticator/fido2/executor/FIDO2Executor.java

```diff
@@ -0,0 +1,210 @@
+
+        String displayName = (String) flowExecutionContext.getFlowUser().getClaim(DISPLAY_NAME_CLAIM_URL);
+        // If the displayName is not available, build the displayName with firstName and lastName.
+        if (StringUtils.isBlank(displayName)) {
+            String firstName = (String) flowExecutionContext.getFlowUser().getClaim(FIRST_NAME_CLAIM_URL);
+            String lastName = (String) flowExecutionContext.getFlowUser().getClaim(LAST_NAME_CLAIM_URL);
+            if (StringUtils.isNotBlank(firstName) || StringUtils.isNotBlank(lastName)) {
+                displayName = StringUtils.join(new String[]{firstName, lastName}, " ");
+            } else {
+                // If the firstName or the lastName is not available, set the username as the displayName.
+                displayName = flowExecutionContext.getFlowUser().getUsername();
+            }
+        }
+        return StringUtils.trim(displayName);
+    }
+
+    private String resolveOrigin(FlowExecutionContext context) {
+
+        String origin = context.getUserInputData().get(FIDO2ExecutorConstants.ORIGIN);
+        if (StringUtils.isNotBlank(origin)) {
+            context.getProperties().put(FIDO2ExecutorConstants.ORIGIN, origin);
+        } else {
+            origin = (String) context.getProperties().get(FIDO2ExecutorConstants.ORIGIN);
+        }
+        return origin;
+    }
+
+    private boolean isInitiation(FlowExecutionContext context) {
+
+        String credential = context.getUserInputData().get(FIDO2ExecutorConstants.CREDENTIAL);
+        return StringUtils.isBlank(credential);
+    }
+
+    private ExecutorResponse clientInputRequiredResponse(ExecutorResponse response, String... fields) {
+
+        response.setResult(Constants.ExecutorStatus.STATUS_CLIENT_INPUT_REQUIRED);
+        response.setRequiredData(Arrays.asList(fields));
+        return response;
+    }
+
+    private ExecutorResponse errorResponse(ExecutorResponse response, String message) {
+
+        response.setErrorMessage(message);
+        response.setResult(Constants.ExecutorStatus.STATUS_ERROR);
+        return response;
+    }
+
+    @Override
+    public List<String> getInitiationData() {
+
+        return Arrays.asList(FIDO2ExecutorConstants.ORIGIN, USERNAME_CLAIM);
+    }
+
+    @Override
+    public ExecutorResponse rollback(FlowExecutionContext context) {
+
+        String credentialId = (String) context.getProperties().get(FIDO2ExecutorConstants.CREDENTIAL_ID);
+        ExecutorResponse response = new ExecutorResponse();
+        if (StringUtils.isNotBlank(credentialId) &&
+                StringUtils.isNotBlank(context.getFlowUser().getUsername())) {
+            try {
+                webAuthnService.deregisterFIDO2Credential(credentialId, context.getFlowUser().getUsername());
+            } catch (FIDO2AuthenticatorServerException | FIDO2AuthenticatorClientException e) {
+                return errorResponse(new ExecutorResponse(), e.getMessage());
```

Comment:
The stack trace will be getting missed hence better to log exception and handled it